src/main/java/Duke.java

```diff
@@ -1,10 +1,228 @@
+//            File inputFile = new File("text-ui-test/input.txt");
+            //Manage input from user
+            Scanner sc = new Scanner(System.in);
+            while (servingDuke.canRun()) {
+                String input = sc.nextLine();
+                servingDuke.respondToInput(input);
+            }
+            sc.close();
+            servingDuke.saveData(fw);
+        } catch (FileNotFoundException e) {
+            System.out.println("The file could not be found.");
+            e.printStackTrace();
+        } catch (IOException e) {
+            System.out.println("Something went wrong.");
+            e.printStackTrace();
+        } catch (DateTimeParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void saveData(FileWriter fw) throws IOException {
+        for (Task t : taskList) {
+            fw.write(t.generateSaveFormat()+ "\n");
+//            System.out.println(t.generateSaveFormat());
+        }
+        fw.close();
+    }
+
+    private void retrieveFromDatabase(File fileDatabase) throws FileNotFoundException {
+        Scanner s = new Scanner(fileDatabase);
+        while (s.hasNext()) {
+            Task next = generateTask(s.nextLine());
+//            System.out.println(next);
+            taskList.add(next);
+        }
+        s.close();
+    }
+
+    private Task generateTask(String nextLine) {
+        String[] taskCharacteristics = nextLine.split(line);
+//        for (int i = 0; i < taskCharacteristics.length; i++) {
+//            System.out.println(i + ": " + taskCharacteristics[i]);
+//        }
+        boolean isDone = taskCharacteristics[1].equals("1");
+        String desc = taskCharacteristics[2];
+        switch (taskCharacteristics[0]) {
+            case ("T"):
+                return new Todo(desc);
+            case ("D"):
+                return new Deadline(desc, LocalDate.parse(taskCharacteristics[3]));
+            case ("E"):
+                return new Event(desc, LocalDate.parse(taskCharacteristics[3]));
+            default:
+                return new Task(desc);
+        }
+    }
+
+    private void respondToInput(String rawInput) {
+        //Remove leading and trailing whitespaces
+        String input = rawInput.trim();
+        String[] breakdown = input.split(" ", 2);
+        if (!input.isEmpty()) {
+            if (input.equals("bye")) {
+                running = false;
+                System.out.println(
+                        constructMessage(farewell));
+            } else if (input.equals("list")) {
+                System.out.println(
+                        constructMessage(task_read + printAsString(taskList)));
+            } else if (breakdown[0].equals("delete")) {
+                deleteTask(breakdown);
+            } else if (breakdown[0].equals("done")) {
+                markTaskAsDone(breakdown);
+            } else {
+                addTask(breakdown);
+            }
+        }
+    }
```

Comment:
Perhaps you can abstract this logic (and other related logic) to another class (eg. Parser)? To build better abstraction.